### ISSUE TITLE
Use Create React App or Next.js on frontend

### ISSUE THREAD

**Issue by fernandoporazzi** [Reckless-Satoshi/robosats#101]
**Is your feature request related to a problem? Please describe.**
The way the front-end code is organised makes it quite hard to install the project and run it locally. Also, by using custom dev and build script, we bring unneeded complexity to our scope. Also, I see that it's quite hard to update dependencies the way it is now. 

**Describe the solution you'd like**
I propose here the usage of [Create React Ap](https://create-react-app.dev/) for the front-end application. It runs solely on top of Javascript and there is no need to have any back-end tool(Ie.: Python or so). CRA also supports some nice tools out of the box, such as instant reloads when you save your code, optimised bundles and etc... Also, `dev`, `test` and `build` script are already build making the developer experience much better.

**Other alternatives**
Another possible alternative here would be the usage of [Next.js](https://nextjs.org/) which also has some nice and exciting features out of the box, pretty much the same as CRA, but also with Server-side rendering, rehydrations and custom server capabilities may we need those. 

**TL;DR**
The way I see the project is organised now, it's gonna be hard to improve current features and also to build new ones.  I suggest the usage of [Create React App](https://create-react-app.dev/) or [Next.js](https://nextjs.org/) for better UI/UX, maintainability and developer experience.

**Comment by Reckless-Satoshi** [comment]
Hello @fernandoporazzi 

Thank you very much for bringing up this issue and your contributions. 

I do not have previous experience with CRA or Next.js . A priori, it seems it would indeed make easier the dev experience, and if anything was not possible while using CRA I read that the project can be ejected (under the hood uses Babel, webpack,...). 

As my frontend developing experience is very limited I am not sure what the trade-offs are, so I need some research a bit more. If you think it is easy to port the project and start using CRA/Next.js, let's do it! :)

_**A little bit off-topic:**_
I was on my way to explore how to create a React Native App that has a single Webview and use react-native-tor as well (more info https://github.com/Reckless-Satoshi/robosats/issues/42 ). However, if jumping into CRA/Next.js requires a more profound rework, this might be the right moment to actually make a migration of the app to a framework that allows seamless web+android+ios development. As I said, I am a total newbie. I have read little about React native, but it seem it could serve this purpose as it also works on the web using [react-native-web](https://necolas.github.io/react-native-web/)
